db/migrate/20200708112930_create_user_accesses.rb

```diff
@@ -0,0 +1,16 @@
+class CreateUserAccesses < ActiveRecord::Migration[5.2]
+  def change
+    create_table :accesses, id: :uuid do |t|
+      t.references :user, null: false, index: true, type: :uuid, foreign_key: true
+      t.string :role, null: false, index: true
+
+      t.references :resource,
```

Comment:
Would be really nice to make this required.  I think we can, it would mean introducing the [Region model](https://github.com/simpledotorg/simple-server/pull/1163) were the root node is an `application` which owns `organizations`.  Then a `super_admin` has `admin` rights on the `application`, and we don't need any special case logic anywhere! Everything would flow through the tree nicely.

Reply:
Yup, let's see how that pans out, we can always do that later depending on how we release things. The change shouldn't be very complicated.